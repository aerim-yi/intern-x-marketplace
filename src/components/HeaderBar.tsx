import { Navbar, Container } from 'react-bootstrap';


const HeaderBar = () => {
    return (
        <Navbar bg="dark" variant="dark">
            <Container>
                <Navbar.Brand href="#home">Intern X</Navbar.Brand>
            </Container>
        </Navbar>
    )
}

export default HeaderBar