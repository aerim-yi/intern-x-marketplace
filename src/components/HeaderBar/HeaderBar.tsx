import { Navbar, Container, ButtonGroup, Button } from 'react-bootstrap';
import { Link, useHistory } from "react-router-dom";
import { useWalletHook } from '../NavBar/useWallethook';
import { Hamburger } from '../HumburgerMenu/HumburgerMenu';

export const HeaderBar: React.FC = () => {
    const history = useHistory();
    const wallet = useWalletHook();

    return (
        <Navbar bg="dark" variant="dark">
            <Container>
                <Hamburger open={true}/>

                <Link to={'/'} style={{ textDecoration: 'none', color: 'black' }}>
                    <Navbar.Brand>Intern X Marketplace</Navbar.Brand>
                </Link>
                {wallet?.walletInfo?.address &&
                    <ButtonGroup>
                        <Link to={'/asset'} style={{ textDecoration: 'none', color: 'black' }}>
                            <Button variant="outline-info">Asset</Button>
                        </Link>
                        <Link to={'/wallet_management'} style={{ textDecoration: 'none', color: 'black' }}>
                            <Button variant="outline-info">Wallet</Button>
                        </Link>
                        <Button variant="outline-info" onClick={() => { wallet?.logout(); history.push('/') }}>
                            Disconnect
                        </Button>
                    </ButtonGroup>
                }
                {!wallet?.walletInfo?.address &&
                    <ButtonGroup className='walletButtonGroup'>
                        <Button variant="outline-info" onClick={wallet?.login}>
                            Connect Wallet
                        </Button>
                    </ButtonGroup>
                }
            </Container>
        </Navbar>
    )
}
