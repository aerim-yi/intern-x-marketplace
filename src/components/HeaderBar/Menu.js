import React from "react";
import "./styles.css";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import { ReactComponent as Logo } from "./logo.svg";

import "bootstrap/dist/css/bootstrap.min.css";

export default function App() {
    return (
        <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
            <Navbar.Brand href="#home">
                <Logo
                    alt=""
                    width="30"
                    height="30"
                    className="d-inline-block align-top"
                />
                Home Page
            </Navbar.Brand>
            <Navbar.Toggle aria-controls="responsive-navbar-nav" />
            <Navbar.Collapse id="responsive-navbar-nav">
                <Nav className="mr-auto">
                    <Nav.Link href="#features">About Us</Nav.Link>
                    <Nav.Link href="#pricing">Language</Nav.Link>
                    <NavDropdown title="Dropdown" id="collasible-nav-dropdown">
                        <NavDropdown.Item href="#action/3.1">Action</NavDropdown.Item>
                        <NavDropdown.Item href="#action/3.2">
                            Help
                        </NavDropdown.Item>
                        <NavDropdown.Item href="#action/3.3">Support</NavDropdown.Item>
                        <NavDropdown.Divider />
                        <NavDropdown.Item href="#action/3.4">
                            Separated link
                        </NavDropdown.Item>
                    </NavDropdown>
                </Nav>
                <Nav>
                    <Nav.Link href="#deets">More info</Nav.Link>
                    <Nav.Link eventKey={2} href="#contact">
                        Contact us
                    </Nav.Link>
                </Nav>
            </Navbar.Collapse>
        </Navbar>
    );
}
