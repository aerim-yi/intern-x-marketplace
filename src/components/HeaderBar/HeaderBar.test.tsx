import { render} from '@testing-library/react'
import HeaderBar from './HeaderBar'

describe('NavBar component', () => {
    test('match snapshot', () => {
        const { container } = render(<HeaderBar />);
        expect(container).toMatchSnapshot()
    })
})