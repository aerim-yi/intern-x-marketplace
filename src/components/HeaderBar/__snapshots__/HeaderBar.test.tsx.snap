// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Intern X Marketplace
      </a>
      <div
        class="walletButtonGroup btn-group"
        role="group"
      >
        <button
          class="btn btn-outline-info"
          type="button"
        >
          Connect Wallet
        </button>
      </div>
    </div>
  </nav>
</div>
`;
